File: user/src/main/java/com/fx/user/exception/handler/UserExceptionHandler.java
package com.fx.user.exception.handler;

import com.fx.global.api.Api;
import com.fx.global.api.ErrorCodeIfs;
import com.fx.user.exception.UserException;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

@Slf4j
@ControllerAdvice
public class UserExceptionHandler {

    @ExceptionHandler(UserException.class)
    public ResponseEntity<Api<ErrorCodeIfs>> handleUserException(UserException e) {
        log.error("",e);
        return Api.ERROR(e.getErrorCodeIfs());
    }


}

File: user/src/main/kotlin/com/fx/user/exception/handler/UserExceptionHandler.kt
package com.fx.user.exception.handler

import com.fx.global.api.Api
import com.fx.global.api.ErrorCodeIfs
import com.fx.user.exception.UserException
import org.slf4j.LoggerFactory
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class UserExceptionHandler {

    private val log = LoggerFactory.getLogger(UserExceptionHandler::class.java)

    @ExceptionHandler(UserException::class)
    fun handleUserException(e: UserException): ResponseEntity<Api<ErrorCodeIfs>> {
        log.error("", e)
        return Api.ERROR(e.errorCodeIfs)
    }

}